src/client.ts

```diff
@@ -346,6 +346,14 @@ export class Client extends EventEmitter {
     this.addInateEventHandlers(stanzaInstance);
     this.proxyStanzaEvents(stanzaInstance);
     stanzaInstance.pinger = new Ping(this, stanzaInstance);
+
+    // handle any extension configuration
+    for (const extension of this.extensions) {
+      if (extension.configureNewStanzaInstance) {
+        await extension.configureNewStanzaInstance(stanzaInstance);
```

Comment:
so this is the solution I came up with for the race condition in spigot. What was happening is spigot was connecting an instance of the sdk then immediately trying to start a screen share session. The problem was some of the config stuff was getting hung up waiting for the ice servers to come back, more specifically the custom stanza definitions. This is ultimately what was causing my spigot problems because it was sending a malformed `mediaUpgrade` presence stanza.